executor/src/witgen/query_processor.rs

```diff
@@ -56,50 +57,82 @@ where
+        match query {
+            Expression::Tuple(items) => Ok(items
                 .iter()
-                .find_map(|MatchArm { pattern, value }| {
-                    (match pattern {
-                        MatchPattern::CatchAll => true,
-                        MatchPattern::Pattern(pattern) => {
-                            rows.evaluate(pattern).unwrap().constant_value() == Some(v)
-                        }
+                .map(|i| self.interpolate_query(i, rows))
+                .collect::<Result<Vec<_>, _>>()?
+                .join(", ")),
+            Expression::String(s) => Ok(format!(
+                "\"{}\"",
+                s.replace('\\', "\\\\").replace('"', "\\\"")
+            )),
+            Expression::MatchExpression(scrutinee, arms) => {
+                let v = self.evaluate_expression(scrutinee, rows)?;
+                let expr = arms
+                    .iter()
+                    .find_map(|MatchArm { pattern, value }| {
+                        (match pattern {
+                            MatchPattern::CatchAll => true,
+                            MatchPattern::Pattern(pattern) => {
+                                self.evaluate_expression(pattern, rows) == Ok(v)
+                            }
+                        })
+                        .then_some(value)
                     })
-                    .then_some(value)
-                })
-                .ok_or(IncompleteCause::NoMatchArmFound)?;
-            interpolate_query(expr, rows)
+                    .ok_or(IncompleteCause::NoMatchArmFound)?;
+                self.interpolate_query(expr, rows)
+            }
+            _ => self.evaluate_expression(query, rows).map(|v| v.to_string()),
+        }
+    }
+
+    fn evaluate_expression(
+        &self,
+        expr: &'a Expression<T>,
+        rows: &RowPair<T>,
+    ) -> Result<T, IncompleteCause<&'a PolynomialReference>> {
+        match expr {
+            Expression::Number(n) => Ok(*n),
+            Expression::BinaryOperation(left, op, right) => Ok(evaluate_binary_operation(
+                self.evaluate_expression(left, rows)?,
+                *op,
+                self.evaluate_expression(right, rows)?,
+            )),
+            Expression::UnaryOperation(op, expr) => Ok(evaluate_unary_operation(
+                *op,
+                self.evaluate_expression(expr, rows)?,
+            )),
+            Expression::Reference(Reference::LocalVar(i, _name)) => {
+                assert!(*i == 0);
+                Ok(rows.current_row_index.into())
+            }
+            Expression::Reference(Reference::Poly(poly)) => {
+                if !poly.next && poly.index.is_none() {
+                    let poly_id = poly.poly_id();
+                    match poly_id.ptype {
+                        PolynomialType::Committed | PolynomialType::Intermediate => {
+                            Ok(rows.get_value(poly).unwrap())
```

Comment:
The `vm_to_vm` test fails here because the the value is not available.

I think in this particular might actually be fixed by #714, because it seems like the `match main_vm.pc {  }` query (on a witness column of the sub-VM) is executed from the main VM, which is fixed by #714.

I still think we shouldn't unwrap here, because we might be able to have the value the next time we process queries.